src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -12099,7 +12099,7 @@ protected override LocalFunctionState CreateLocalFunctionState(LocalFunctionSymb
         {
             var variables = (symbol.ContainingSymbol is MethodSymbol containingMethod ? _variables.GetVariablesForMethodScope(containingMethod) : null) ??
                 _variables.GetRootScope();
-            return new LocalFunctionState(LocalState.ReachableStateWithNotNulls(variables));
+            return new LocalFunctionState(LocalState.StateWithNotNulls(variables, reachable: false));
```

Comment:
>variables [](http://example.com/codeflow?start=71&length=9)

Why do we need to drag variables into this? Can we simply create a regular unreachable state? What would be the difference?

Reply:
"Regular unreachable state" would not have any effect since it doesn't "contain" any variables but only contained variables are considered from "local function starting state"s when we copy them into "local function body analysis state" here:

https://github.com/dotnet/roslyn/blob/6389afe618316a310d20b1f6915ffc8bc6fbd172/src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs#L3034-L3050

Effectively, all variables would be considered "maybe null". But we want to consider captured variables as "not null".